am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -136,83 +137,72 @@ public void revokeResourceAccess(@NotNull @Valid ExplicitAccessMetadata accessMe
      */
-    @SuppressWarnings("PMD") // AvoidLiteralsInIfCondition: magic number used until multiple roles are supported
-    @AuditLog("filtered access to resource '{{resource.resourceId}}' "
-        + "defined as '{{resource.type.serviceName}}|{{resource.type.resourceType}}|{{resource.type.resourceName}}' "
-        + "for accessor '{{userId}}' in roles '{{userRoles}}': {{result.permissions}}")
-    public FilterResourceResponse filterResource(@NotBlank String userId,
-                                                 @NotEmpty Set<@NotBlank String> userRoles,
-                                                 @NotNull @Valid Resource resource) {
-        if (userRoles.size() > 1) {
-            throw new IllegalArgumentException("Currently a single role only is supported. "
-                + "Future implementations will allow for multiple roles.");
-        }
-
+    @AuditLog("filtered access to resource '{{resource.resourceId}} defined as '{{resource}} for accessor '{{userId}}' "
+        + "in roles '{{userRoles}}': {{result.access.permissions}}")
+    public FilteredResourceEnvelope filterResource(@NotBlank String userId,
+                                                   @NotEmpty Set<@NotBlank String> userRoles,
+                                                   @NotNull @Valid Resource resource) {
         List<ExplicitAccessRecord> explicitAccess = jdbi.withExtension(AccessManagementRepository.class,
             dao -> dao.getExplicitAccess(userId, resource.getResourceId()));
 
         Map<JsonPointer, Set<Permission>> attributePermissions;
+        AccessType accessManagementType;
 
         if (explicitAccess.isEmpty()) {
-            List<RoleBasedAccessRecord> roleBasedAccess = jdbi.withExtension(AccessManagementRepository.class,
-                dao -> dao.getRolePermissions(
-                    resource.getType().getServiceName(),
-                    resource.getType().getResourceType(),
-                    resource.getType().getResourceName(),
-                    userRoles.iterator().next()));
-
-            if (roleBasedAccess.isEmpty()) {
-                return null;
-            }
+            attributePermissions = getRolePermissions(resource.getType(), userRoles);
 
-            if (explicitAccessType(userRoles)) {
+            if (attributePermissions == null) {
                 return null;
             }
 
-            attributePermissions = roleBasedAccess.stream().collect(getMapCollector());
+            accessManagementType = AccessType.ROLE_BASED;
 
         } else {
             attributePermissions = explicitAccess.stream().collect(getMapCollector());
+            accessManagementType = AccessType.EXPLICIT;
         }
 
         JsonNode filteredJson = filterService.filterJson(resource.getResourceJson(), attributePermissions);
 
-        return FilterResourceResponse.builder()
+        return FilteredResourceEnvelope.builder()
             .resourceId(resource.getResourceId())
```

Comment:
I think we should rename:
- `resource.getResourceId()` into `resource.getId()`
- `resource.getType()` into `resource.getDefinition()`
- `resource.getResourceJson()` into `resource.getData()`

Reply:
It might also make sense to have the following as next step:

```
@Data
@Builder
public final class Resource {
    @NotBlank
    private final String id;
    @NotNull
    @Valid
    private final ResourceDefinition definition;
    @NotNull
    private final JsonNode data;
}

@Data
@Builder
public final class FilteredResourceEnvelope {
    private Resource resource;
    private AccessEnvelope access;
}
```

Reason is that since envelope contains all properties from resource it make sense to reuse resource definition.